File: StrategyPattern/bank/ConcreteWithdraw2.java
package bank;

public class ConcreteWithdraw2 implements WithdrawStrategy
{
	public void withdraw(Account acc, int money)
	{
		int balance = acc.getBalance();
		int bound = ((MinusAccount)acc).getBound();
		
		if(balance + bound >= money)
		{
			balance -= money;
			acc.setBalance(balance);
			System.out.println("Withdraw Money : " + money + ", Balance : " + balance);
		}
		else
		{
			System.out.println("Exceeding of Limit!");
		}

	}
}

File: Strategy/C++/ConcreteWithdraw2.cpp
#include "ConcreteWithdraw2.h"
#include "MinusAccount.h"


ConcreteWithdraw2::ConcreteWithdraw2()
{
}

ConcreteWithdraw2::~ConcreteWithdraw2()
{
}

int ConcreteWithdraw2::withdraw(Account* acc, int money)
{
	int balance = acc->getBalance();

	if (balance + ((MinusAccount*)acc)->getBound() >= money)
	{
		balance -= money;
		cout << "Withdraw Money : " << money << ", Balance : " << balance << endl;
	}
	else
	{
		cout << "Exceeding of Limit!" << endl;
	}

	return balance;
}
